Documentation/pre-built-detection.md

```diff
@@ -0,0 +1,147 @@
+# Pre-built detection guide
+
+This is a detailed guide on how to approach enabling pre-built detection for a source-buildable repository.
+It is primarily intended for repository maintainers in the `dotnet` organization.
```

Comment:
Per my previous comment this should target developers contributing to the dotnet project.

Reply:
Agreed, reworked